chia/wallet/wallet_state_manager.py

```diff
@@ -1417,17 +1399,32 @@ async def _add_coin_states(
                             additions = [state.coin for state in children]
                             if len(children) > 0:
                                 fee = 0
-
+                                known_puzzle_hash: bool = await self.puzzle_store.puzzle_hash_exists(
+                                    coin_state.coin.puzzle_hash
+                                )
                                 to_puzzle_hash = None
+                                coin_spend: Optional[CoinSpend] = None
                                 # Find coin that doesn't belong to us
                                 amount = 0
-                                for coin in additions:
+                                for child_state in children:
+                                    coin: Coin = child_state.coin
                                     derivation_record = await self.puzzle_store.get_derivation_record_for_puzzle_hash(
                                         coin.puzzle_hash
                                     )
                                     if derivation_record is None:  # not change
                                         to_puzzle_hash = coin.puzzle_hash
                                         amount += coin.amount
+                                        if coin_spend is None:
+                                            coin_spend = await fetch_coin_spend_for_coin_state(coin_state, peer)
```

Comment:
Why is all this code related to the spend processing here in the loop?

Reply:
This code path is for the coin that is already spent the first time we saw it. This usually means we are doing a resync. But since we are adding the clawback coin to interested_coin manually for the sender, it cannot recover during the resync. This means during the resync, the sender will not receive any coin state about the Clawback coin. So we want to get the coin spends for these coins and check if they are Clawback coins owned by the current wallet.

Reply:
Unless we change the hint mechanism fundamentally, there isn't another way. Currently, we only check the first element in the memo and see if it is a known puzzle hash. We need to check all elements in the hint. But this may cause other issues.